### PR TITLE
perlre: Clarify /xx pod

### PR DESCRIPTION
**Pull request by khwilliamson**
This is a result of GH #19726

### CODE REVIEW

File: pod/perlre.pod [hunk 1]
```diff
@@ -492,6 +492,10 @@ may be easier to grasp than the squashed equivalents
     /[d-eg-i3-7]/
     /[!@"#$%^&*()=?<>']/
 
+Note that this unfortunately doesn't mean that your bracketed classes
+can extend over multiple lines.  The closing bracket must be on the same
+line as the opening one.
```
Comment (hvds) [inline pod/perlre.pod]: That reads like it's a syntax error, but it actually just means newline will be one of the characters in the class. I think the second sentence could read something like: "Unless the closing bracket is on the same line as the opening one, the newline character will be part of the class just as if you'd written `C<\n>`." (Sadly that also means we'd have back-compat problems if we wanted it to ignore newlines in the future.)

The placement of the `/xx` docs directly after "A common pitfall is to forget that `"#"` characters begin a comment under `/x` ..." may also explain some of the additional confusion - the parenthetical "... (outside a bracketed character class, which is unaffected by `/x`) ..." a couple of paragraphs earlier can easily get lost here. It's not immediately obvious to me how to rearrange things to minimize the opportunity for such confusion.

**Comment by khwilliamson** [comment]
On 5/16/22 18:12, Hugo van der Sanden wrote:
> Unless the closing bracket is on the same line as the opening one, the 
> newline character will be part of the class just as if you'd written 
> |C<\n>|.

Much better than what I wrote


I repushed with much more detail

File: pod/perlre.pod [hunk 2]
```diff
@@ -474,15 +474,18 @@ You can use L</(?#text)> to create a comment that ends earlier than the
...
+character under C</x>.
 
 Starting in Perl v5.26, if the modifier has a second C<"x"> within it,
-it does everything that a single C</x> does, but additionally
-non-backslashed SPACE and TAB characters within bracketed character
-classes are also generally ignored, and hence can be added to make the
-classes more readable.
+it increases the effect of a single C</x>.  The only difference is that
+inside bracketed character classes, non-backslashed SPACE and TAB
+characters classes are also generally ignored, and hence can be added to
```
Comment (hvds) [inline pod/perlre.pod]: `s/classes //`

**Review by hvds — APPROVED** [review]
LGTM other than the spurious word.

I'd love to see ".. are also generally ignored" replaced with something more prescriptive some time, but I don't think it urgent to do as part of this PR.

File: pod/perlre.pod (hunk 2 above)
Comment (hvds) [reply pod/perlre.pod]: ^^ that was line 487 "characters classes"

**Pull request merged**